File: 0242.cs
// URL: https://leetcode.com/problems/valid-anagram/                        
// TITLE: Valid Anagram                            
// DIFFICULTY: Easy                                
// ------------------------------------------------------
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length){ return false; }
        Dictionary<char, int> dict = new Dictionary<char, int>();
        foreach (char i in s){
            if (dict.ContainsKey(i)){
                dict[i] += 1;
            }
            else {
                dict.Add(i, 1);
            }
        }
        foreach (char i in t){
            if (dict.ContainsKey(i)){
                if (dict[i] == 0) {return false;}
                dict[i] -= 1;
            }
            else {
                return false;
            }
        }
        return true;
    }
}

File: 0242.py
# URL: https://leetcode.com/problems/valid-anagram/                        
# TITLE: Valid Anagram                            
# DIFFICULTY: Easy                                
# ------------------------------------------------------
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t): return False
        unique = set(s)
        if set(t) != unique: return False

        for letter in unique:
            if s.count(letter) != t.count(letter): return False

        return True